ArmoniK.SDK.Client/src/SessionServiceImpl.cpp

```diff
@@ -351,51 +301,40 @@ void SessionServiceImpl::CleanupTasks(const std::set<std::string> &task_ids) {
   // Cancel the given tasks
   while (tasks_iterator != task_ids.end()) {
     channel_pool.WithChannel([&](const std::shared_ptr<::grpc::Channel> &channel) {
-      auto stub = armonik::api::grpc::v1::tasks::Tasks::NewStub(channel);
-      ::grpc::ClientContext context;
-      armonik::api::grpc::v1::tasks::CancelTasksRequest request;
+      std::vector<std::string> batched_ids;
       for (size_t i = 0; i < batch_size && tasks_iterator != task_ids.end(); ++i) {
-        *request.mutable_task_ids()->Add() = *tasks_iterator;
+        batched_ids.push_back(*tasks_iterator);
         tasks_iterator++;
       }
-      armonik::api::grpc::v1::tasks::CancelTasksResponse response;
-      auto status = stub->CancelTasks(&context, request, &response);
-      if (!status.ok()) {
-        throw ArmoniK::Sdk::Common::ArmoniKSdkException("Unable to cancel tasks " + status.error_message());
-      }
+      armonik::api::client::TasksClient(armonik::api::grpc::v1::tasks::Tasks::NewStub(channel))
+          .cancel_tasks(batched_ids);
     });
   }
 
   tasks_iterator = task_ids.begin();
 
   while (tasks_iterator != task_ids.end()) {
-    armonik::api::grpc::v1::tasks::GetResultIdsResponse response;
     // List batch of results from the given tasks
-    channel_pool.WithChannel([&](const std::shared_ptr<::grpc::Channel> &channel) {
-      auto stub = armonik::api::grpc::v1::tasks::Tasks::NewStub(channel);
-
-      ::grpc::ClientContext context;
-      armonik::api::grpc::v1::tasks::GetResultIdsRequest request;
+    auto map_results = channel_pool.WithChannel([&](const std::shared_ptr<::grpc::Channel> &channel) {
+      std::vector<std::string> batched_ids;
       for (size_t i = 0; i < batch_size && tasks_iterator != task_ids.end(); ++i) {
-        *request.mutable_task_id()->Add() = *tasks_iterator;
+        batched_ids.push_back(std::move(*tasks_iterator));
```

Comment:
The code could be more relevant using `emplace_back` here, don't you think @lemaitre-aneo ?

Reply:
It would actually be harder to read to read because we could not be sure just seeing this line what happens to the vector (see my other comment).

tldr; use `push_back` for copy or move existing elements, use `emplace_back` to _construct_ a new object in situ